src/config.rs

```diff
@@ -284,15 +287,14 @@ macro_rules! create_config {
         }
 
         impl PartialConfig {
-            pub fn to_toml(&self) -> Result<String, String> {
+            pub fn to_toml(&self) -> Result<String, Error> {
```

Comment:
iiuc, the preferred way to use Failure is to use the concrete error type in the lib, and only use the `Error` type in an app, or when a function can have multiple kinds of error